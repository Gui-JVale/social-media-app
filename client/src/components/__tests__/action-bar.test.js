import React from 'react';
import { unmountComponentAtNode } from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Cta from '../general/cta/cta.component';
import ActionBar from '../general/action-bar/action-bar.component';

describe('<ActionBar />', () => {
  let wrapper = null;
  let container = null;
  let tree = null;

  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);

    // setup component for test
    wrapper = shallow(
      <ActionBar>
        <Cta text="test1" />
        <Cta text="test2" />
        <Cta text="test3" />
      </ActionBar>,
    );
    tree = renderer
      .create(
        <ActionBar>
          <Cta text="test1" />
          <Cta text="test2" />
          <Cta text="test3" />
        </ActionBar>,
      )
      .toJSON();
  });

  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it('Renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
