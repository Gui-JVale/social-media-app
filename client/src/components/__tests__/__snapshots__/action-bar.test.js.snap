// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionBar /> Renders correctly 1`] = `
<div
  className="action-bar"
>
  <div
    className="action-bar__container u-margin-top-medium"
  >
    <div
      className="cta"
    >
      <span
        className="cta__text"
      >
        test1
      </span>
    </div>
    <div
      className="cta"
    >
      <span
        className="cta__text"
      >
        test2
      </span>
    </div>
    <div
      className="cta"
    >
      <span
        className="cta__text"
      >
        test3
      </span>
    </div>
  </div>
</div>
`;
